<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>propMovement</class>
 <widget class="QWidget" name="propMovement">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>200</width>
    <height>422</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>200</width>
    <height>130</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QGroupBox" name="a_groupAnimation">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Animation</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <property name="spacing">
       <number>6</number>
      </property>
      <property name="leftMargin">
       <number>6</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>6</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QPushButton" name="playButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <kerning>true</kerning>
         </font>
        </property>
        <property name="text">
         <string>‚èµ</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="currentFrameSlider">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="b_groupTree">
     <property name="minimumSize">
      <size>
       <width>200</width>
       <height>0</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="title">
      <string>Sequention</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>6</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>6</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QTreeView" name="treeView">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>150</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>300</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>7</pointsize>
         </font>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="sizeAdjustPolicy">
         <enum>QAbstractScrollArea::AdjustToContents</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="c_groupFrame">
     <property name="title">
      <string>Frame</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QWidget" name="a_time_widget" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>6</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>6</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="timeLabel">
           <property name="text">
            <string>Show time:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="timeSpinBox"/>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="b_keys_widget" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>48</height>
         </size>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>6</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>6</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="matrixLabel">
           <property name="text">
            <string>Transf. matrix:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="clearButton">
           <property name="maximumSize">
            <size>
             <width>24</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="toolTip">
            <string>clear transformation</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset resource="dpVision.qrc">
             <normaloff>:/icons/Erase.ico</normaloff>:/icons/Erase.ico</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="copyButton">
           <property name="maximumSize">
            <size>
             <width>24</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="toolTip">
            <string>copy matrix to clipboard</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset resource="dpVision.qrc">
             <normaloff>:/icons/Copy.ico</normaloff>:/icons/Copy.ico</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>16</width>
             <height>16</height>
            </size>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pasteButton">
           <property name="maximumSize">
            <size>
             <width>24</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="toolTip">
            <string>paste matrix from clipboard</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset resource="dpVision.qrc">
             <normaloff>:/icons/Paste.ico</normaloff>:/icons/Paste.ico</iconset>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="c_matrix_widget" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QTableWidget" name="matrixTable">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>182</width>
             <height>82</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>182</width>
             <height>82</height>
            </size>
           </property>
           <property name="font">
            <font>
             <pointsize>7</pointsize>
            </font>
           </property>
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Sunken</enum>
           </property>
           <property name="verticalScrollBarPolicy">
            <enum>Qt::ScrollBarAlwaysOff</enum>
           </property>
           <property name="horizontalScrollBarPolicy">
            <enum>Qt::ScrollBarAlwaysOff</enum>
           </property>
           <property name="autoScroll">
            <bool>false</bool>
           </property>
           <property name="dragDropOverwriteMode">
            <bool>false</bool>
           </property>
           <property name="textElideMode">
            <enum>Qt::ElideNone</enum>
           </property>
           <property name="gridStyle">
            <enum>Qt::SolidLine</enum>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
           <property name="cornerButtonEnabled">
            <bool>false</bool>
           </property>
           <property name="rowCount">
            <number>4</number>
           </property>
           <property name="columnCount">
            <number>4</number>
           </property>
           <attribute name="horizontalHeaderVisible">
            <bool>false</bool>
           </attribute>
           <attribute name="horizontalHeaderMinimumSectionSize">
            <number>45</number>
           </attribute>
           <attribute name="horizontalHeaderDefaultSectionSize">
            <number>45</number>
           </attribute>
           <attribute name="horizontalHeaderHighlightSections">
            <bool>false</bool>
           </attribute>
           <attribute name="verticalHeaderVisible">
            <bool>false</bool>
           </attribute>
           <attribute name="verticalHeaderMinimumSectionSize">
            <number>20</number>
           </attribute>
           <attribute name="verticalHeaderDefaultSectionSize">
            <number>20</number>
           </attribute>
           <attribute name="verticalHeaderHighlightSections">
            <bool>false</bool>
           </attribute>
           <row/>
           <row/>
           <row/>
           <row/>
           <column/>
           <column/>
           <column/>
           <column/>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="dpVision.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>clearButton</sender>
   <signal>clicked()</signal>
   <receiver>propMovement</receiver>
   <slot>clearMatrix()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>127</x>
     <y>427</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>414</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>copyButton</sender>
   <signal>clicked()</signal>
   <receiver>propMovement</receiver>
   <slot>copyToClipboard()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>151</x>
     <y>427</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>414</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pasteButton</sender>
   <signal>clicked()</signal>
   <receiver>propMovement</receiver>
   <slot>pasteFromClipboard()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>175</x>
     <y>427</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>414</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>treeView</sender>
   <signal>activated(QModelIndex)</signal>
   <receiver>propMovement</receiver>
   <slot>onTreeViewItemClicked(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>99</x>
     <y>184</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>currentFrameSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>propMovement</receiver>
   <slot>onSliderValueChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>114</x>
     <y>25</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playButton</sender>
   <signal>clicked()</signal>
   <receiver>propMovement</receiver>
   <slot>onPlayButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>18</x>
     <y>25</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>changedEul(double)</slot>
  <slot>changedTra(double)</slot>
  <slot>changedSca(double)</slot>
  <slot>changedQua(double)</slot>
  <slot>clearMatrix()</slot>
  <slot>copyToClipboard()</slot>
  <slot>pasteFromClipboard()</slot>
  <slot>onRotButton()</slot>
  <slot>onScaleCheck(int)</slot>
  <slot>onTreeViewItemClicked(QModelIndex)</slot>
  <slot>onSliderValueChanged(int)</slot>
  <slot>onPlayButtonClicked()</slot>
 </slots>
</ui>
